lib/connection.js

```diff
@@ -1099,28 +1099,28 @@ Connection.prototype.loginBySoap = function(username, password, callback) {
 };
 
 /**
- * Logout the current session
- *
- * Setting revoke to true will revoke API access
+ * Logout the current session 
  *
+ * @param {Boolean} revoke - Revokes API Access if set to true
  * @param {Callback.<undefined>} [callback] - Callback function
  * @returns {Promise.<undefined>}
  */
-Connection.prototype.logout = function(callback, revoke = false) {
+Connection.prototype.logout = function(revoke = false, callback) {
```

Comment:
The change will break current API client which uses `Connection#logout(callback)` without revoke parameter.
Please check the first parameter is function or not, like `Connection#request()` does for the options.
https://github.com/jsforce/jsforce/blob/132f5823178d34524feed4fde075f31ad4751b96/lib/connection.js#L294-L298
Also you cannot use default parameter in ES5.

Reply:
Good catch, updated in d125c3d